### ISSUE TITLE
Sharing: accessibilty issues with options screen

### ISSUE THREAD

**Issue by jeherve** [Automattic/jetpack#2709]
In #185, we made some changes to the sharing options screen to make it more accessible. Since we've made those changes, one can use their keyboard to do the following:
1. Move buttons from one area to another by hitting Enter. Selected button will then cycle from "Available", to "Enabled", to "Hidden", and then back to "Available".
2. Change the order of the buttons in one area thanks to the arrow keys.

Unfortunately, it is still very difficult to manage sharing buttons using a screen reader.

There are 2 basic things we could do to improve that screen's accessibility:
- [ ] Provide a role for each one of the available buttons. When you focus on one of these buttons with a screen reader, it doesn’t identify them as either buttons or links. Folks consequently have no way to know they can actually hit Enter to move the buttons from one area to another.
- [ ] When hitting "Enter", provide feedback to the screen reader to make it clear that the button moved to a new area.

Another alternative would be to provide an alternative interface with checkboxes and/or select buttons, much like Core's accessibility mode for the widgets.

It's probably also worth considering #141 when making that decision; whatever we do, it would be nice for the settings to become mobile-friendly as well.

@davidakennedy Do you have any ideas about this?

**Comment by davidakennedy** [comment]
I do have a few ideas... let me take a closer look, and get back to you.

I'd love to see us steer clear of an alternate interface. It's more code, and more complicated. There should be a way we can make this work for everyone without impacting design or creating two designs.

**Comment by jeherve** [comment]
Closing this for now since we're now developing a new Publicize settings screen that will be responsive and live under Jetpack > Settings.

**Comment by richardmtl** [comment]
Requested in 3032096-t
